PlayerCollection.cs

```diff
@@ -200,18 +207,22 @@ private void UpdateDetailsButtonWithNewCharacter()
             _detailsButton.BasicTooltipText = GetPreviousCharactersToolTipText();
         }
 
-        private Dropdown CreateDropdown()
+        private Dropdown CreateDropdown(Image roleIcon)
         {
             var dropdown = new Dropdown
             {
                 Parent = _detailsButton,
-                Width = 150
+                Width = 130
             };
             dropdown.Items.Add(_placeholderRoleName);
             foreach (var role in _availableRoles.OrderBy(role => role.Name.ToLowerInvariant()))
             {
                 dropdown.Items.Add(role.Name);
             }
+            dropdown.ValueChanged += delegate
+            {
+                roleIcon.Texture = _availableRoles.FirstOrDefault(role => role.Name.Equals(dropdown.SelectedItem))?.Icon ?? null;
```

Comment:
Nitpick: (to disregard if you wrote it that way for explicitness) no need for the `?? null` here as the `?.` already returns null if no role is found.